1.data-exploration/scripts/nbconverted/5.apply_pca.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Apply PCA to gene dependencies
+# 
+# Principal component analysis (PCA) is a commonly-used dimensionality reduction method that finds components that explain orthogonal variation in the data in a descending fashion.
+# 
+# We apply it to GeneEffect scores here and save latent representations (PCA components) for downstream comparative analyses (to compare with BetaVAE).
+
+# In[1]:
+
+
+import sys
+import pathlib
+import pandas as pd
+import plotnine as gg
+from sklearn.decomposition import PCA
+
+sys.path.insert(0, "../0.data-download/scripts/")
+from data_loader import load_train_test_data
+
+
+# In[2]:
+
+
+data_directory = pathlib.Path("../0.data-download/data")
+dependency_file = pathlib.Path(f"{data_directory}/CRISPRGeneEffect.csv")
+gene_dict_file = pathlib.Path(f"{data_directory}/CRISPR_gene_dictionary.tsv")
+
+output_dir = pathlib.Path("results")
+pca_output_file = pathlib.Path(f"{output_dir}/pca_latent.csv.gz")
+output_pca_weights_file = pathlib.Path(f"{output_dir}/PCA_weight_matrix_gsea.csv")
+
+
+# In[3]:
+
+
+pca_components = 50
```

Comment:
Is there a way to avoid hardcoding and obtain the latent dimensions from `3.beta_vae_training_extract_model_components` ? Maybe this file can be moved to 2.train-VAE section to make that easier and since it's products are not needed until `3.analysis`.

Reply:
> Is there a way to avoid hardcoding and obtain the latent dimensions from 3.beta_vae_training_extract_model_components ?

The betaVAE and PCA are not related, except for the fact that they both perform compression and use the same input data. We can, theoretically identify the same number of PCs as the number of gene features, which would remove hardcoding, but I think it is ok for this preliminary analysis.

> Maybe this file can be moved to 2.train-VAE section to make that easier and since it's products are not needed until 3.analysis.

This is a great idea. I will move!